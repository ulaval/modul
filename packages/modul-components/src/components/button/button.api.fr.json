{
    "props": {
        "type": {
            "description": "Type de bouton"
        },
        "skin": {
            "description": "Propriété pour changer le visuel du bouton"
        },
        "disabled": {
            "description": "Propriété pour désactiver le bouton"
        },
        "waiting": {
            "description": "Propriété pour mettre le bouton en attente"
        },
        "fullSize": {
            "description": "Propriété pour mettre le bouton pleine largeur"
        },
        "iconName": {
            "description": "Nom de l'icône"
        },
        "iconPosition": {
            "description": "Propriété pour positioner l'icône"
        },
        "iconSize": {
            "description": "Propriété pour changer la grosse de l'icône"
        },
        "precision": {
            "description": "Ajouter un texte de précision au bouton"
        }
    },
    "slots": {
        "default": {
            "description": "Contenu du bouton"
        },
        "precision": {
            "description": "Contenu du block precision du bouton"
        }
    }
}
