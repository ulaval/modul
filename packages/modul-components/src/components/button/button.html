<button class="m-button"
        :class="{ 'm--is-skin-primary': isSkinPrimary,
               'm--is-skin-secondary': isSkinSecondary,
               'm--is-skin-tertiary': isSkinTertiary,
               'm--is-waiting': isWaiting,
               'm--is-disabled': disabled,
               'm--is-full-size': fullSize }"
        :disabled="disabled || isWaiting"
        :type="type"
        @click="onClick"
        @mousedown="onMousedown"
        @focus="onFocus"
        @blur="onBlur"
        @touchstart="onTouchstart"
        @touchend="onTouchend">
    <m-icon v-if="hasIconLeft"
            aria-hidden="true"
            class="m-button__icon m--is-left"
            :name="iconName"
            :size="iconSize"
    />
    <m-spinner v-if="hasWaitingIconLeft"
               aria-hidden="true"
               class="m-button__icon m--is-left"
               skin="lighter"
               size="small"
    />
    <span class="m-button__text">
        <slot />
        <span class="m-button__precision"
              v-if="hasPrecision">
            <slot name="precision"></slot>{{precision}}
        </span>
    </span>
    <m-spinner v-if="hasWaitingIconRight"
               aria-hidden="true"
               class="m-button__icon m--is-right"
               skin="lighter"
               size="small"
    />
    <m-icon v-if="hasIconRight"
            aria-hidden="true"
            class="m-button__icon m--is-right"
            :name="iconName"
            :size="iconSize"
    />
</button>
