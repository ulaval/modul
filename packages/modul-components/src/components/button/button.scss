@import 'commons';

$m-button--border-radius: 4px !default;

.m-button {
    position: relative;
    display: inline-flex;
    align-items: center;
    text-align: center;
    vertical-align: middle;
    font-size: inherit;
    border-radius: $m-button--border-radius;
    padding: 10px $m-space;
    font-weight: $m-font-weight-semi-bold;
    min-height: 44px;

    @include m-button();

    @include m-is-ie() {
        display: inline;
    }

    &:not(.m--is-waiting):not(.m--is-disabled) {
        &.m--is-skin-primary {
            @include m-button--primary();
        }

        &.m--is-skin-secondary {
            @include m-button--secondary();
        }

        &.m--is-skin-tertiary {
            @include m-button--tertiary();
        }

        @include m-button--hover();
    }

    &.m--is-waiting {
        @include m-button--waiting();
    }

    &.m--is-disabled {
        @include m-button--disabled();
    }

    &.m--is-full-size {
        width: 100%;
    }

    .m-icon {
        stroke-width: 2px;

        @include m-font-size('xs');
    }

    &__text,
    &__precision {
        position: relative;
    }

    &__precision,
    &__icon {
        &:empty {
            display: none;
        }
    }

    &__text {
        flex: 1;
    }

    &__icon {
        position: relative;
        z-index: 2;
        font-size: inherit;
        height: #{$m-font-size * 1px};

        &.m--is-left {
            margin-right: $m-space-sm
        }

        &.m--is-right {
            margin-left: $m-space-sm;
        }
    }

    &__precision {
        display: block;
        margin-top: 3px;
        font-weight: $m-font-weight-light;

        @include m-font-size('sm');

        &:empty {
            display: none;
        }
    }

    &__hidden {
        @include m-visually-hidden;
    }
}
