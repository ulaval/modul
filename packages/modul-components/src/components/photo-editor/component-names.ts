export const CROP_IMAGE_NAME: string = 'm-crop-image';
export const IMAGE_DISPLAYER_NAME: string = 'm-image-displayer';
