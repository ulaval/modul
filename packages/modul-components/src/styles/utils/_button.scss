.mu-button-primary {
    @include m-button();
    @include m-button--primary();
    @include m-button--hover();
}

.mu-button-secondary {
    @include m-button();
    @include m-button--secondary();
    @include m-button--hover();
}

.mu-button-disabled {
    @include m-button();
    @include m-button--disabled();
}

.mu-button-waiting {
    @include m-button();
    @include m-button--waiting();
}
