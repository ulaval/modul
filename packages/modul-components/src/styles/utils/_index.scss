@import 'utils.mixin';

@include mu-utils();
