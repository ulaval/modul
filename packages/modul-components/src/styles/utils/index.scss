@import 'typographies';
@import 'content';
@import 'layout';
@import 'visually-hidden';
