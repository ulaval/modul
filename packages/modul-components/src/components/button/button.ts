import Vue, { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Emit, Prop } from 'vue-property-decorator';
import { Enums } from '../../utils/enums/enums';
import { BUTTON_NAME } from '../component-names';
import { MIcon } from '../icon/icon';
import { MSpinner } from '../spinner/spinner';
import WithRender from './button.html?style=./button.scss';

export enum MButtonType {
    Button = 'button',
    Submit = 'submit',
    Reset = 'reset'
}

export enum MButtonSkin {
    Primary = 'primary',
    Secondary = 'secondary',
    Tertiary = 'tertiary'
}

export enum MButtonIconPosition {
    Left = 'left',
    Right = 'right'
}

@WithRender
@Component({
    components: {
        MSpinner,
        MIcon
    }
})
export class MButton extends Vue {
    @Prop({
        default: MButtonType.Button,
        validator: value => Enums.toValueArray(MButtonType).includes(value)
    })
    public readonly type!: MButtonType;

    @Prop({
        default: MButtonSkin.Primary,
        validator: value => Enums.toValueArray(MButtonSkin).includes(value)
    })
    public readonly skin!: MButtonSkin;

    @Prop()
    public readonly precision: string;

    @Prop({ default: false })
    public readonly disabled!: boolean;

    @Prop({ default: false })
    public readonly waiting!: boolean;

    @Prop({ default: false })
    public readonly fullSize!: boolean;

    @Prop()
    public readonly iconName?: string;

    @Prop({
        default: MButtonIconPosition.Left,
        validator: value => Enums.toValueArray(MButtonIconPosition).includes(value)
    })
    public readonly iconPosition!: MButtonIconPosition;

    @Prop({ default: '12px' })
    public readonly iconSize: string;

    @Emit('click')
    public onClick(event: Event): void { }

    @Emit('mousedown')
    public onMousedown(event: MouseEvent): void { }

    @Emit('touchstart')
    public onTouchstart(event: TouchEvent): void { }

    @Emit('touchend')
    public onTouchend(event: TouchEvent): void { }

    @Emit('focus')
    public onFocus(event: FocusEvent): void { }

    @Emit('blur')
    public onBlur(event: FocusEvent): void { }

    public get isSkinPrimary(): boolean {
        return this.skin === MButtonSkin.Primary;
    }

    public get isSkinSecondary(): boolean {
        return this.skin === MButtonSkin.Secondary;
    }

    public get isSkinTertiary(): boolean {
        return this.skin === MButtonSkin.Tertiary;
    }

    public get isWaiting(): boolean {
        return !this.disabled ? this.waiting : false;
    }

    public get hasIcone(): boolean {
        return Boolean(this.iconName);
    }

    public get hasIconLeft(): boolean {
        return this.iconPosition === MButtonIconPosition.Left && this.hasIcone && !this.waiting;
    }

    public get hasIconRight(): boolean {
        return this.iconPosition === MButtonIconPosition.Right && this.hasIcone && !this.waiting;
    }

    public get hasWaitingIconLeft(): boolean {
        return this.iconPosition === MButtonIconPosition.Left && this.waiting;
    }

    public get hasWaitingIconRight(): boolean {
        return this.iconPosition === MButtonIconPosition.Right && this.waiting;
    }

    public get hasPrecision(): boolean {
        return !!this.precision || !!this.$slots.precision;
    }
}

const ButtonPlugin: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug(BUTTON_NAME, 'plugin.install');
        v.component(BUTTON_NAME, MButton);
    }
};

export default ButtonPlugin;
