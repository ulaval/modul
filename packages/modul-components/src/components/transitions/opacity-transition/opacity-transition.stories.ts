import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../../conf/storybook/utils';
import { MButtonSkin } from '../../button/button';
import { OPACITY_TRANSITION_NAME } from '../../component-names';
import OpacityTransitionPlugin from './opacity-transition';

Vue.use(OpacityTransitionPlugin);

storiesOf(`${componentsHierarchyRootSeparator}${OPACITY_TRANSITION_NAME}`, module)

    .add('Default', () => ({
        data: () => ({
            open: true,
            disabledTransition: false,
            buttonSkin: MButtonSkin.Secondary,
            transitionEmit: ''
        }),
        methods: {
            toggleDisplay(): void {
                let _this: any = this as any;
                _this.open = !_this.open;
            },
            toggleDisabledTransition(): void {
                let _this: any = this as any;
                _this.disabledTransition = !_this.disabledTransition;
            },
            beforeEnter(): void {
                (this as any).transitionEmit = 'before-enter';
                this.$log.log(`@emit('before-enter')`);
            },
            enter(): void {
                (this as any).transitionEmit = 'enter';
                this.$log.log(`@emit('enter')`);
            },
            afterEnter(): void {
                (this as any).transitionEmit = 'after-enter';
                this.$log.log(`@emit('after-enter')`);
            },
            enterCancelled(): void {
                (this as any).transitionEmit = 'enter-cancelled';
                this.$log.log(`@emit('enter-cancelled')`);
            },
            beforeLeave(): void {
                (this as any).transitionEmit = 'before-leave';
                this.$log.log(`@emit('before-leave')`);
            },
            leave(): void {
                (this as any).transitionEmit = 'leave';
                this.$log.log(`@emit('leave')`);
            },
            afterLeave(): void {
                (this as any).transitionEmit = 'after-leave';
                this.$log.log(`@emit('after-leave')`);
            },
            leaveCancelled(): void {
                (this as any).transitionEmit = 'leave-cancelled';
                this.$log.log(`@emit('leave-cancelled')`);
            }
        },
        template: `
        <div>
            <div class="m-u--margin-bottom">
                <m-button @click="toggleDisplay()">
                    {{open ? 'Hide' : 'Show'}} icon
                </m-button>
                <m-button class="m-u--margin-left"
                          :skin="buttonSkin"
                          @click="toggleDisabledTransition()">
                    {{disabledTransition ? 'Enabled' : 'Disabled'}} transition
                </m-button>
            </div>
            <m-opacity-transition :disabled="disabledTransition"
                                  @before-enter="beforeEnter"
                                  @enter="enter"
                                  @after-enter="afterEnter"
                                  @enter-cancelled="enterCancelled"
                                  @before-leave="beforeLeave"
                                  @leave="leave"
                                  @after-leave="afterLeave"
                                  @leave-cancelled="leaveCancelled">
                <m-icon v-if="open" name="m-svg__clock"></m-icon>
            </m-opacity-transition>
            <p v-if="transitionEmit">Last transition emit: {{transitionEmit}}</p>
        </div>`
    }))
    .add('zoom-effect="false"', () => ({
        data: () => ({
            open: true
        }),
        methods: {
            toggleDisplay(): void {
                let _this: any = this as any;
                _this.open = !_this.open;
            }
        },
        template: `
        <div>
            <div>
                <m-button @click="toggleDisplay()">
                    {{open ? 'Hide' : 'Show'}} icon
                </m-button>
            </div>
            <m-opacity-transition :zoom-effect="false">
                <m-icon v-if="open" class="m-u--margin-top" name="m-svg__clock"></m-icon>
            </m-opacity-transition>
        </div>`
    }));
