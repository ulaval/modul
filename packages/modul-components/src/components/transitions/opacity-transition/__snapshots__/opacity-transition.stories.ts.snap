// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components|m-opacity-transition Default 1`] = `
<div>
  <div
    class="m-u--margin-bottom"
  >
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        
                    Hide icon
                 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
     
    <button
      class="m-button m-u--margin-left m--is-skin-secondary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        
                    Disabled transition
                 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
  </div>
   
  <svg
    aria-hidden="true"
    class="m-icon"
    height="20px"
    width="20px"
  >
    <!---->
     
    <use
      aria-hidden="true"
      class=""
    />
  </svg>
   
  <!---->
</div>
`;

exports[`Storyshots components|m-opacity-transition zoom-effect="false" 1`] = `
<div>
  <div>
    <button
      class="m-button m--is-skin-primary"
      type="button"
    >
      <!---->
       
      <!---->
       
      <span
        class="m-button__text"
      >
        
                    Hide icon
                 
        <!---->
      </span>
       
      <!---->
       
      <!---->
    </button>
  </div>
   
  <svg
    aria-hidden="true"
    class="m-icon m-u--margin-top"
    height="20px"
    width="20px"
  >
    <!---->
     
    <use
      aria-hidden="true"
      class=""
    />
  </svg>
</div>
`;
